jumpscale/sals/process/__init__.py

```diff
@@ -714,32 +1014,79 @@ def get_memory_usage():
 def get_environ(pid):
     """Gets env vars for a specific process based on pid
 
-    Arguments:
-        pid (int) -- process pid
+    Args:
+        pid (int): process pid
+
+    Raises:
+        psutil.NoSuchProcess: if process with the given PID is not found and die set to True
+        psutil.AccessDenied: if permission denied
+
+    Returns:
+        dict: dict of env variables
+    """
+    proc = get_process_object(pid)
+    try:
+        return proc.environ()
+    except (psutil.AccessDenied, psutil.NoSuchProcess) as e:
+        raise e
+
+
+def kill_proc_tree(
+    parent, sig=signal.SIGTERM, include_parent=True, include_grand_children=True, timeout=5, sure_kill=False
+):
+    """Kill a process and its children (including grandchildren) with signal `sig`
+
+    Args:
+        proc (int or psutil.Process): Target process ID (PID) or psutil.Process object.
+        sig (signal, optional): See signal module constants. Defaults to signal.SIGTERM.
+        include_parent (): Whether to kill the process itself. Defaults to True.
+        include_grand_children (): whether to kill recursively all grandchildren. Defaults to True.
+        timeout (int, optional): How long to wait for a process to terminate (seconds) before raise exception\
+            or, if sure_kill=True, send a SIGKILL. Defaults to 5.
+        sure_kill (bool, optional): Whether to fallback to SIGKILL if the timeout exceeded for the terminate operation or not. Defaults to False.
 
     Returns:
-        [dict] -- dict of env variables
+        (list[psutil.Process] or None): list of process objects that remain alive if any. None otherwise.
```

Comment:
The same as previous comment to return an empty list instead of `None`.

Reply:
The same as a previous replay on similar comments :D